lib/services/role.go

```diff
@@ -948,6 +948,16 @@ func (set RoleSet) EnumerateDatabaseUsers(database types.Database, extraUsers ..
 	return result
 }
 
+// GetAllowedKubeUsersAndGroupsForCluster works on a given role set to return a minimal list of
+// configured kubernetes_users and kubernetes_groups for a given kubernetes cluster.
+// The wildcard selector is ignored, since it is not allowed for kubernetes principals.
+func (set RoleSet) GetAllowedKubeUsersAndGroupsForCluster(kube types.KubeCluster) (kubeUsers []string, kubeGroups []string) {
+	matcher := NewKubernetesClusterLabelMatcher(kube.GetAllLabels())
+	// ignore error since we are only interested in Allowed  groups
+	kubeGroups, kubeUsers, _ = set.CheckKubeGroupsAndUsers(5*time.Minute, true /* force ttl override*/, matcher)
```

Comment:
I don't think we want to hardcode the TTL here. It should be a parameter to `GetAllowedKubeUsersAndGroupsForCluster` and should be populated based on user identity's expiration.

Reply:
My thought was: since users have no control over the UI session duration, it is not possible for them to change the value to be able to receive `kubernetes_users' and `kubernetes_groups' from roles whose MaxSessionTTL is less than the session expiration.  So I wanted to not validate the TTL (override=true) so that they can receive all the groups they can eventually use if their session is less than the MaxSessionTTL value for that role. Otherwise, we could just return values that might not include some groups, which would be strange to the user since they were configured.